File: redsheet-portal/src/app/database-models/v-nano-discussion-sequence-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vNanoDiscussionSequenceRecord {
    constructor(vNanoDiscussionSequenceRecord?: vNanoDiscussionSequenceRecord )  {
        let properties = [ "nanoDiscussionSequenceId", "negotiationPhaseId", "process", "projectModuleId", "sortOrder", "tacticId", "time",];
        for (let property of properties)
            if (vNanoDiscussionSequenceRecord && vNanoDiscussionSequenceRecord[property])
                this[property] = vNanoDiscussionSequenceRecord[property];
    }
 
	nanoDiscussionSequenceId : number =   0 ;
 
    @range(1,2147483647)
	negotiationPhaseId : number =   undefined;
 
    @required()
    @maxLength(1000)
	process : string =   undefined;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
    @range(1,2147483647)
	sortOrder : number =   undefined;
 
    @range(1,2147483647)
	tacticId : number =   undefined;
 
    @required()
	time : Date =   undefined;


}

File: RedSheet.DbEntities/Models/vNanoDiscussionSequenceRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vNanoDiscussionSequenceRecords", Schema = "dbo")]
	[Serializable]
	public partial class vNanoDiscussionSequenceRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int NanoDiscussionSequenceId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int NegotiationPhaseId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Process { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int SortOrder { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TacticId { get; set; } 
 
	[Required]
	public TimeSpan Time { get; set; } 

	public vNanoDiscussionSequenceRecord()
	{
    }

    }
}

